jwst/datamodels/schemas/referencefile.schema.yaml

```diff
@@ -1,32 +1,200 @@
+            fits_keyword: P_PUPIL
+          pupil_imaging_lens:
+            title: Pupil imaging lens in the optical path?
+            type: boolean
+            fits_keyword: PILIN
+          grating:
+            title: Name of the grating element used
+            type: string
+            enum: [G140M, G235M, G395M, G140H, G235H, G395H, PRISM, MIRROR, N/A, ANY]
+            fits_keyword: GRATING
+          band:
+            title: MRS wavelength band
+            type: string
+            enum: [SHORT, MEDIUM, LONG, N/A, ANY]
+            fits_keyword: BAND
+          fixed_slit:
+            title: Name of fixed slit aperture used
+            type: string
+            fits_keyword: FXD_SLIT
+          focus_position:
+            title: Focus position
+            type: number
+            fits_keyword: FOCUSPOS
+          coronagraph:
+            title: coronagraph mask used
+            type: string
+            enum: [4QPM, LYOT, MASK210R, MASKSWB, MASK335R, MASK430R, MASKLWB, NONE]
+            fits_keyword: CORONMSK
+      exposure:
+        title: Exposure parameters
+        type: object
+        properties:
+          p_type:
+            title: Type of data in the exposure
+            type: string
+            pattern: "|\
+              ^FGS_(?=IMAGE|DARK|FOCUS|INTFLAT|SKYFLAT)|\
+              ^MIR_(?=IMAGE|TACQ|LYOT|4QPM|LRS-FIXEDSLIT|LRS-SLITLESS|MRS|DARK|FLAT-IMAGE|FLATIMAGE|FLAT-MRS|FLATMRS|CORONCAL)|\
+              ^NIS_(?=AMI|DARK|FOCUS|IMAGE|LAMP|SOSS|TACQ|TACONFIRM|WFSS)|\
+              ^NRC_(?=IMAGE|GRISM|TACQ|CORON|FOCUS|DARK|FLAT|LED|WFSC|TACONFIRM|TSIMAGE|TSGRISM)|\
+              ^NRS_(?=AUTOFLAT|AUTOWAVE|BOTA|BRIGHTOBJ|CONFIRM|DARK|FIXEDSLIT|FOCUS|IFU|IMAGE|LAMP|MIMF|MSASPEC|TACONFIRM|TACQ|TASLIT)|\
+              ANY|N//A"
+            fits_keyword: P_EXPTYP
+      subarray:
+        title: Subarray parameters
+        type: object
+        properties:
+          name:
+            title: Subarray used
+            type: string
+            pattern: "|\
+              1024X16|128X128|128X2048|2048X128|2048X64|32X32|\
+              64X2048|8X8|ALLSLITS|BRIGHTSKY|FULL|GENERIC|\
+              MASKA210R|MASKA335R|MASKA430R|MASKBLWB|MASKBSWB|MASK1065|\
+              MASK1140|MASK1550|MASKLYOT|\
+              S1600A1|S200A1|S200A2|S200B1|S400A1|SLITLESSPRISM|STRIPE|\
+              SUB1024A|SUB1024B|SUB128|SUB128LLCORNER|SUB128DIAGONAL|\
+              SUB128CENTER|SUB16|SUB160|SUB160P|SUB1A|SUB1B|\
+              SUB2048|SUB256|SUB32|SUB32LLCORNER|SUB32DIAGONAL|SUB32CENTER|\
+              SUB320|SUB400P|SUB512|SUB64|SUB640|\
+              SUB64P|SUB80|SUB8LLCORNER|SUB8DIAGONAL|SUB8CENTER|SUB96|\
+              SUBAMPCAL|SUBFP1A|SUBFP1B|SUBGRISM128|\
+              SUBGRISM256|SUBGRISM64|SUBSTRIP256|SUBSTRIPE256|SUBSTRIP96|\
+              SUBPRISM|SUBSTRIP80|SUBTASOSS|SUBTAAMI|\
+              WFSS128C|WFSS128R|WFSS64C|WFSS64R|N/A
+              "
+            fits_keyword: P_SUBARR
```

Comment:
This comment is to document what @hbushouse and I discussed which is - although EXP_TYPE and P_EXPTYP are not used for selection EXP_TYPE can be used to describe the type of observation used to create the reference file.